core/src/main/java/org/apache/iceberg/data/GenericAppenderFactory.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *  *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.iceberg.data;
+
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.MetricsConfig;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.avro.Avro;
+import org.apache.iceberg.data.avro.DataWriter;
+import org.apache.iceberg.data.orc.GenericOrcWriter;
+import org.apache.iceberg.data.parquet.GenericParquetWriter;
+import org.apache.iceberg.deletes.EqualityDeleteWriter;
+import org.apache.iceberg.deletes.PositionDeleteWriter;
+import org.apache.iceberg.encryption.EncryptedOutputFile;
+import org.apache.iceberg.io.FileAppender;
+import org.apache.iceberg.io.FileAppenderFactory;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.orc.ORC;
+import org.apache.iceberg.parquet.Parquet;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Map;
+
+/** Copied from https://github.com/apache/iceberg
+ * Line 223: add metricsConfig for ORC PositionDeleteWriter in newPosDeleteWriter method
```

Comment:
Can you add some comment to tell why we need add this metrics config for orc writer?

Reply:
BTW, as you are using the same full class name with iceberg class, how to ensure that the program uses the class we want in case of class conflicts?

Reply:
> Can you add some comment to tell why we need add this metrics config for orc writer?

add metricsConfig for ORC PositionDeleteWriter in newPosDeleteWriter method to ensure get metrics accurately, such as lowerBounds and upperBounds in com.netease.arctic.io.TestTaskWriter#testBasePosDeleteWriter, https://github.com/NetEase/amoro/blob/master/core/src/test/java/com/netease/arctic/io/TestTaskWriter.java#L141

Reply:
> BTW, as you are using the same full class name with iceberg class, how to ensure that the program uses the class we want in case of class conflicts?

As far as I know,  It will use org.apache.iceberg.data.GenericAppenderFactory in Amoro project instead of external libraries  during compile and buld. 

The test phenomenon in com.netease.arctic.io.TestTaskWriter#testBasePosDeleteWriter also meets expectations

Reply:
We should try to avoid modifying Iceberg's code directly unless there is a necessary reason. Perhaps we can modify the unit test cases to have different behavior when using the ORC format?

Reply:
> We should try to avoid modifying Iceberg's code directly unless there is a necessary reason. Perhaps we can modify the unit test cases to have different behavior when using the ORC format?

Thanks,  I had changed unit com.netease.arctic.io.TestTaskWriter#testBasePosDeleteWriter, skip to check lower bounds and upper bounds for orc